lib/datasets.go

```diff
@@ -858,6 +859,24 @@ func (datasetImpl) List(scope scope, p *ListParams) ([]dsref.VersionInfo, error)
 	return infos, nil
 }
 
+func getProfile(ctx context.Context, pros profile.Store, idStr, peername string) (pro *profile.Profile, err error) {
+	if idStr == "" {
+		// TODO(b5): we're handling the "me" keyword here, should be handled as part of
+		// request scope construction
+		if peername == "me" {
+			return pros.Owner(), nil
+		}
+		return profile.ResolveUsername(pros, peername)
+	}
+
+	id, err := profile.IDB58Decode(idStr)
+	if err != nil {
+		log.Debugw("decoding profile ID", "err", err)
+		return nil, err
+	}
+	return pros.GetProfile(id)
+}
+
```

Comment:
This used to live in `lib/profile.go`, but now those methods are exclusive to the active user. Moved this closer to it's only caller.